test/tests/lock_tests.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import print_function, unicode_literals, division
+
+import grp
+from pavilion import lockfile
+import os
+import subprocess as sp
+import sys
+import time
+import unittest
+
+
+# NOTE: The lockfile class is designed to work over NFS, but these tests don't actually check for
+#  that.
+class TestConfig(unittest.TestCase):
+    lock_path = 'lock_test.lock'
+
+    def setUp(self):
+        if os.path.exists(self.lock_path):
+            print("\nRemoving lockfile {} from old (failed) run.".format(self.lock_path),
+                  file=sys.stderr)
+            os.unlink(self.lock_path)
+
+    def test_locks(self):
+
+        # Make sure the lock can be created and deleted with no contention and in non-blocking mode.
+        with lockfile.LockFile(self.lock_path):
+            # Make sure the lock was created.
+            self.assertTrue(os.path.exists(self.lock_path))
+        # Make sure the lock is deleted after close.
+        self.assertFalse(os.path.exists(self.lock_path))
+
+        lockfile.LockFile._create_lockfile(self.lock_path, 100, '1234')
+
+        # Remove the lockfile after 1 second of trying.
+        sp.call("sleep 1; rm {}".format(self.lock_path), shell=True)
+        # Test waiting for the lockfile.
+        with lockfile.LockFile(self.lock_path, timeout=2):
+            pass
+
+        # Making sure that we can automatically acquire and delete an expired lockfile.
+        lockfile.LockFile._create_lockfile(self.lock_path, -100, '1234')
+        with lockfile.LockFile(self.lock_path, timeout=1):
+            pass
+
+        # Lock objects are reusable.
+        lock = lockfile.LockFile(self.lock_path)
+        with lock:
+            pass
+        with lock:
+            pass
+
+        os.getgid()
```

Comment:
A lovely function call, but I'm not sure it is performing any... function.

Reply:
I'm not sure what you're referring to.

Reply:
You call os.getgid(), but the value isn't assigned to anything, so it seems useless.

Reply:
I was going to check that the group got set reasonably. I think I forgot to actually write that test.